### PR TITLE
add document use composer extra assets

### PR DESCRIPTION
**Pull request by alavieille**

### CODE REVIEW

File: en/developer_guide/assets_bower_npm.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+Manage Bower and Npm Dependencies
+=================================
+
+Open Orchestra using npm to manage some server side javascript libraries and
```
Comment (ngilain) [inline en/developer_guide/assets_bower_npm.rst]: Open Orchestra uses

File: en/developer_guide/assets_bower_npm.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+bower to manage the client side libraries.
+
+To ease adding bower and npm dependencies, Open Orchestra uses the composer plugin
+`composer-extra-assets`_ which allows to put dependencies in the composer.json
```
Comment (ngilain) [inline en/developer_guide/assets_bower_npm.rst]: which allows to declare such dependencies

**Comment by SOLenG** [comment]
I was wondering if it should not add a link in the section "install asstes" to your new documentation.
Otherwise :+1:

File: en/developer_guide/assets_bower_npm.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Adding Bower and Npm Dependencies
+---------------------------------
+
+To add your own Bower and Npm dependencies you have to specify in composer.json of your application or your bundle.
```
Comment (ngilain) [inline en/developer_guide/assets_bower_npm.rst]: you have to specify them in the composer.json of your

File: en/developer_guide/assets_bower_npm.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
+Manage Bower and Npm Dependencies
+=================================
+
+Open Orchestra using npm to manage some server side javascript libraries and
+bower to manage the client side libraries.
+
+To ease adding bower and npm dependencies, Open Orchestra uses the composer plugin
```
Comment (nicolasThal) [inline en/developer_guide/assets_bower_npm.rst]: To facilitate the addition of bower and npm deps

**Comment by alavieille** [comment]
@nicolasThal @itkg-ngilain

File: en/developer_guide/assets_bower_npm.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        "koala-framework/composer-extra-assets": "~2.0"
+    },
+    "extra": {
+        "require-npm": {
+             "gulp": "*"
+        },
+        "expose-npm-packages": true
+    }
+
+.. _`composer-extra-assets`: https://github.com/koala-framework/composer-extra-assets
```
Comment (ngilain) [inline en/developer_guide/assets_bower_npm.rst:53]: I think it lacks a paragraph or the one existing should be rewrite to tell the integrator where it is recommanded to declare its dependancy. For instance, is it allowed or bad to play with the bower.json and package.json files for instance ?

**Comment by alavieille** [comment]
@itkg-ngilain

File: en/developer_guide/assets_bower_npm.rst [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+             "gulp": "*"
+        },
+        "expose-npm-packages": true
+    }
+
+Bower.json and Package.json files
+---------------------------------
+
+``composer-extra-assets`` generates ``bower.json`` and ``package.json`` files.
+it's not recommended to add your Bower and Npm dependencies in these files because she will be overwritten
```
Comment (ngilain) [inline en/developer_guide/assets_bower_npm.rst]: It's not necessary
It will be overwritten

**Comment by alavieille** [comment]
@itkg-ngilain

File: en/developer_guide/assets_bower_npm.rst [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        "require-npm": {
+            "grunt": "0.4.*"
+        },
+        "require-bower": {
+            "jquery": "*"
+        },
+    }
+
+If you add Npm dependencies in your bundle, you can add the ``expose-npm-packages``
+attribute to the composer, with this option , the Npm package are available in the node_modules
```
Comment (veolys-nanne) [inline en/developer_guide/assets_bower_npm.rst]: . With this option....

File: en/developer_guide/assets_bower_npm.rst [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        },
+        "expose-npm-packages": true
+    }
+
+Bower.json and Package.json files
+---------------------------------
+
+``composer-extra-assets`` generates ``bower.json`` and ``package.json`` files.
+It's not recommended to add your Bower and Npm dependencies in these files because it will be overwritten
+when the next ``composer update|install``.
```
Comment (veolys-nanne) [inline en/developer_guide/assets_bower_npm.rst]: by the next.... or when the next .... will be launched

**Comment by alavieille** [comment]
@itkg-nanne

**Comment by veolys-nanne** [comment]
:+1:

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**